tests/unit/transaction_freeze_and_bytes_test.py

```
@@ -611,3 +616,92 @@ def test_unsigned_transaction_can_be_signed_after_to_bytes():
+
+def test_transaction_freeze_without_node_ids(mock_client):
+    """
+    Test freeze_with() initializes transaction bytes using clients network node id.
+    """
+    tx = TransferTransaction()
+    tx.freeze_with(mock_client)
+
+    assert tx.node_account_ids == []
+    # Verify creates transaction_bytes for client network nodes
+    assert len(tx._transaction_body_bytes) == len(mock_client.network.nodes)
+    assert set(tx._transaction_body_bytes.keys()) == set(node._account_id for node in mock_client.network.nodes)
```

Ruff (0.14.10)

Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)